float32 x
float32 y
float32 theta
float32[9] covariance